nginx/user_conf.d/default.conf

```diff
@@ -19,21 +43,87 @@ server {
     ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256';
 
     # SSL sessions
-    ssl_session_timeout 1d;
+    ssl_session_timeout 30m;
     # ssl_session_cache dfine in stream and http
     ssl_session_tickets off;
+    ssl_session_cache shared:SSL:30m;
+    # this is the internal Docker DNS, cache only for 30s
+    resolver 127.0.0.11 valid=30s;
+    ssl_stapling on;
+    ssl_stapling_verify on;
+    ssl_buffer_size 8k;
+
+    # Let nginx start if upstream host is unavailable or down
+    set $ckan ckan:5000;
+
+    ##
+    # Nginx Bad Bot Blocker Includes
+    # REPO: https://github.com/mitchellkrogza/nginx-ultimate-bad-bot-blocker
+    ##
+    include /etc/nginx/bots.d/ddos.conf; 
+    include /etc/nginx/bots.d/blockbots.conf;
+
+    gzip on;
+    gzip_comp_level 5;
+    gzip_min_length 1k;
+    gzip_buffers 4 16k;
+    gzip_proxied any;
+    gzip_types
+        application/javascript
+        application/x-javascript
+        text/javascript
+        text/css
+        text/xml
+        application/xhtml+xml
+        application/xml
+        application/atom+xml
+        application/rdf+xml
+        application/rss+xml
+        application/geo+json
+        application/json
+        application/ld+json
+        application/manifest+json
+        application/x-web-app-manifest+json
+        image/svg+xml
+        text/x-cross-domain-policy;
+    gzip_static on;  
+    gzip_disable "MSIE [1-6]\.";
+
+    # For demo.depositar.io CKAN
+    location = / {
+        # Setup language prefix
+        set $prefix_language $http_accept_language;
+        if ($http_accept_language ~* '^(.+?),') {
+            set $prefix_language $1;
+        }
+
+        # Setup redirect Host
+        set $redirect_host $host;
+        if ($http_X_Forwarded_Host != '') {
+            set $redirect_host $http_X_Forwarded_Host;
+        }
 
-    #access_log  /var/log/nginx/host.access.log  main;
+        # Setup URL
+        set $lang_url $scheme://$redirect_host;
+
+        if ($prefix_language !~* 'zh') {
+            return 302 $lang_url/en/;
+        }
+
+        proxy_pass http://$ckan$request_uri;
+        include /etc/nginx/user_conf.d/ckan/depositar_params;
```

Comment:
```suggestion
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_set_header Host $host;
        proxy_read_timeout 600s;
        # when a client closes the connection then keep the channel to uwsgi open. Otherwise uwsgi throws an IOError
        uwsgi_ignore_client_abort on;
```